### PR TITLE
dbeaver/dbeaver#18770 save previous user dialog size

### LINKED ISSUE #18770: Windows sizes/positions of "Connection configuration", "driver properties" not saved
### Description

"Connection configuration", "driver properties" windows parameters not saved.
Each time it opens with default sizes at default positions and needs to be resized a lot to fit everything needed.

I tried ton of parameters combinations with mariadb-connector .jar to make it work with named pipes and each time I must resize windows to get to needed inputs/parameters. Maybe 150-200 times. And it is annoying a lot.

### DBeaver Version

22.3.2

### Operating System

Win 10

### Database and driver

1. Open "Create connection".
2. Resize/move opened window.
3. Press [driver properties] button.
4. Resize/move opened window.
5. Save connection.
6. Open connection... window size/pos is lost.
7. Press [driver properties] button - window size/pos is lost (and it's size is tiny so nothing fits there and window must be resized).

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/dbeaver#18770
The edit connection dialog now should remember the previous user size and try to adjust if it is too small for the current page.

### CODE REVIEW

**Review by ShadelessFox — CHANGES_REQUESTED** [review]
1. Override `Dialog#getInitialSize` instead.
2. You don't need extra props nor listeners. Just `Math.max` of the desired size and saved size.

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 1]
```diff
@@ -97,6 +97,27 @@ protected Control createContents(Composite parent) {
         return contents;
     }
 
+    @Override
+    protected Point getInitialSize() {
+        Point minSize = super.getInitialSize();
+        IDialogSettings settings = getDialogBoundsSettings();
+        Point proposedSize = new Point(minSize.x, minSize.y);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java]: Redundant copy; you can use the returned value of `super.getInitialSize()`

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 2]
```diff
@@ -97,6 +97,27 @@ protected Control createContents(Composite parent) {
...
+            if (width != DIALOG_DEFAULT_BOUNDS) {
+                proposedSize.x = Math.max(width, proposedSize.x);
+            }
+            int height = settings.getInt("DIALOG_HEIGHT");
+            if (height != DIALOG_DEFAULT_BOUNDS) {
+                proposedSize.y = Math.max(height, proposedSize.y);
+            }
+        }
+        catch (NumberFormatException e) {
+        }
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java]:
```suggestion
        } catch (NumberFormatException ignored) {
        }

```

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 3]
```diff
@@ -97,6 +97,27 @@ protected Control createContents(Composite parent) {
...
     }
 
+    @Override
+    protected Point getInitialSize() {
+        Point minSize = super.getInitialSize();
+        IDialogSettings settings = getDialogBoundsSettings();
+        Point proposedSize = new Point(minSize.x, minSize.y);
+        try {
+            // Get the stored width and height.
+            int width = settings.getInt("DIALOG_WIDTH");
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java]: I understand that these constants are defined in the `Dialog` class, but can you please make constants for these?

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 4]
```diff
@@ -49,6 +49,8 @@ public class EditConnectionDialog extends MultiPageWizardDialog {
     private static final Map<DBPDataSourceContainer, EditConnectionDialog> openDialogs = Collections.synchronizedMap(new IdentityHashMap<>());
 
     private static final int TEST_BUTTON_ID = 2000;
+    private static final String PROP_DIALOG_RECOMMENDED_SIZE_HEIGHT = "dialog.preferredSize.height";
+    private static final String PROP_DIALOG_RECOMMENDED_SIZE_WIDTH = "dialog.preferredSize.width";
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java]: These are not used anymore

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 5]
```diff
@@ -97,6 +98,27 @@ protected Control createContents(Composite parent) {
...
+                size.x = Math.max(width, size.x);
+            }
+            int height = settings.getInt(DIALOG_HEIGHT);
+            if (height != DIALOG_DEFAULT_BOUNDS) {
+                size.y = Math.max(height, size.y);
+            }
+        }
+        catch (NumberFormatException ignored) {
+
+        }
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java]:
```suggestion
        } catch (NumberFormatException ignored) {
        }

```

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 6]
```diff
@@ -97,6 +98,26 @@ protected Control createContents(Composite parent) {
...
+            if (width != DIALOG_DEFAULT_BOUNDS) {
+                size.x = Math.max(width, size.x);
+            }
+            int height = settings.getInt(DIALOG_HEIGHT);
+            if (height != DIALOG_DEFAULT_BOUNDS) {
+                size.y = Math.max(height, size.y);
+            }
+        }
+        catch (NumberFormatException ignored) {
+        }
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java]:
```suggestion
        } catch (NumberFormatException ignored) {
        }

```

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Comment by serge-rider** [comment]
Let's avoid hac with internal Eclipse prefs reading.

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/dialogs/MultiPageWizardDialog.java [hunk 7]
```diff
@@ -90,7 +90,9 @@ public MultiPageWizardDialog(IWorkbenchWindow window, IWizard wizard, IStructure
     }
 
     protected Point getInitialSize() {
-        return new Point(700, 500);
+        Point minSize = new Point(700, 500);
+        Point suggestedSize = super.getInitialSize();
+        return new Point(Math.max(minSize.x, suggestedSize.x), Math.max(minSize.y, suggestedSize.y));
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/dialogs/MultiPageWizardDialog.java:95]: There are two unnecessary allocations.

```suggestion
        Point size = super.getInitialSize();
        size.x = Math.min(size.x, 700);
        size.y = Math.min(size.y, 500);
        return size;
```

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java [hunk 8]
```diff
@@ -75,8 +76,7 @@ protected IDialogSettings getDialogBoundsSettings() {
 
     @Override
     protected Control createContents(Composite parent) {
-        Control contents = super.createContents(parent);
-
+        Control contents = super.createContents(parent);;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/EditConnectionDialog.java:79]: Please remove unrelated changes (in other places too).

**Comment by Matvey16** [comment]
Verified

**Pull request merged**